pcdsdevices/lodcm.py

```diff
@@ -60,6 +70,760 @@ def __init__(self, prefix, *args, **kwargs):
+        return ((self.h2n_state.position == 'Si' or
+                 self.h2n_state.position == 'OUT') and
+                self.y2_state.position == 'Si' and
+                self.chi2_state.position == 'Si')
+
+    def get_reflection(self, as_tuple=False, check=False):
+        """
+        Get crystal's reflection.
+
+        Parameters
+        ----------
+        as_tuple : bool
+            Indicates if it should return it as string or tuple format.
+            Defaults to `False`.
+        check : bool
+            Indicates if an exception should be raised in case it could not
+            determine the crystal's reflection. Defaults to `False`.
+
+        Returns
+        -------
+        reflection : str or tuple
+        """
+        reflection = None
+        if self.is_diamond():
+            reflection = self.diamond_reflection.get()
+        elif self.is_silicon():
+            reflection = self.silicon_reflection.get()
+
+        if check and reflection is None:
+            raise ValueError('Unable to determine the crystal reflection')
+        if not as_tuple and reflection is not None:
+            return ''.join(map(str, reflection))
+        return tuple(reflection)
+
+    def get_material(self):
+        """
+        Get the current material.
+
+        Returns
+        -------
+        material : str
+            Material of the crystal.
+
+        Raises
+        ------
+        ValueError
+            When the material could not be determined or is something else
+             other than `Si` or `C`.
+        """
+        if self.is_diamond():
+            return 'C'
+        elif self.is_silicon():
+            return 'Si'
+        else:
+            raise ValueError(
+                "Unable to determine crystal material for Tower 2")
+
+
+class DiagnosticsTower(BaseInterface, Device):
+    """
+    Diagnostic Tower Motors.
+
+    Parameters
+    ----------
+    prefix : str
+        Epics base PV prefix.
+    """
+    # Located midway between T1 and T2 in the center of rotation of the device.
+    # horizontal slits
+    dh = FCpt(IMS, '{self._m_prefix}:MON:MMS:16',
+              kind='normal', doc='LOM Dia H')
+    # vertical slits
+    dv = FCpt(IMS, '{self._m_prefix}:MON:MMS:17',
+              kind='normal', doc='LOM Dia V')
+    dr = FCpt(IMS, '{self._m_prefix}:MON:MMS:19',
+              kind='normal', doc='LOM Dia Theta')
+    # filters wheel
+    df = FCpt(IMS, '{self._m_prefix}:MON:MMS:{self._df_suffix}', kind='normal',
+              doc='LOM Dia Filter Wheel')
+    # pips diode
+    dd = FCpt(IMS, '{self._m_prefix}:MON:MMS:18',
+              kind='normal', doc='LOM Dia PIPS')
+    # yag screen
+    yag_zoom = FCpt(IMS, '{self._m_prefix}:MON:CLZ:01',
+                    kind='normal', doc='LOM Zoom')
+
+    tab_component_names = True
+
+    def __init__(self, prefix, *args, **kwargs):
+        self._m_prefix = ''
+        self._df_suffix = '27'
```

Comment:
The `df` component - (filters wheel i think) - has a different PV suffix in `XPP` vs `XCS` - I should add  a comment in there to explain that

```
XPP:MON:MMS:27
```
vs

```
HFX:MON:MMS:22
```

Reply:
I added a comment about this in fa843ab

Reply:
Sorry, I wasn't very clear. I meant why is the `_m_prefix` handled differently than `_df_prefix`?
As in why is df set to `27` instead of `''` before the `if` with the `self._df_suffix = '27'` moved to inside the if condition

Reply:
Ohh got it, no reason at all, one was set to default to one and the other was not - I should probably change them to have similar implementations